api/router.go

```diff
@@ -19,17 +19,33 @@ func HandleError(s *client.Schemas, t HandleFuncWithError) http.Handler {
 		if err := t(rw, req); err != nil {
 			logrus.WithError(err).Warnf("HTTP handling error")
 
+			statusCode := http.StatusInternalServerError
 			if datastore.ErrorIsNotFound(err) {
-				rw.WriteHeader(http.StatusNotFound)
-				return
+				statusCode = http.StatusNotFound
 			}
-
-			apiContext := api.GetApiContext(req)
-			apiContext.WriteErr(err)
+			writeErr(rw, req, err, statusCode)
 		}
 	}))
 }
 
+func writeErr(rw http.ResponseWriter, req *http.Request, err error, statusCode int) {
+	logrus.WithError(err).Errorf("Error in request")
```

Comment:
The error has been logged by the caller, so here it will cause a redundant log.

Reply:
Update the log message, should be 
```
	if writeErr != nil {
		logrus.WithError(writeErr).Error("Failed to write error %v", err)
	}
```t
The second error is printed only when failing to write error

Reply:
What I mean is `logrus.WithError(err).Errorf("Error in request")` in `writeErr` is not required, since it has been logged in the caller `logrus.WithError(err).Warnf("HTTP handling error")`.

Reply:
Oh, yes. It's redundant. Removed, thank you.